lib/common/appstart/app_initializer.dart

```diff
@@ -55,6 +66,52 @@ Future<void> initializeApp(bool isBackground) async {
     setup.setupScheduleSource();
   }
 
+  // Callback-Function for synchronizing the device calendar with the schedule, when schedule is updated
+  var preferenceProvider = KiwiContainer().resolve<PreferencesProvider>();
+  KiwiContainer()
+      .resolve<ScheduleProvider>()
+      .addScheduleUpdatedCallback((schedule, start, end) async {
+    List<DateEntry> listDateEntries = List<DateEntry>.empty(growable: true);
+    schedule.entries.forEach(
+      (element) {
+        DateEntry date = DateEntry(
+            room: element.room,
+            comment: element.details,
+            databaseName: 'DHBW',
+            description: element.title,
+            year: element.start.year.toString(),
+            start: element.start,
+            end: element.end);
+        listDateEntries.add(date);
+      },
+    );
+    KiwiContainer().resolve<ListDateEntries30d>().listDateEntries =
+        listDateEntries;
+
+    if (await preferenceProvider.isCalendarSyncEnabled()) {
+      Calendar selectedCalendar = await preferenceProvider.getSelectedCalendar();
+      if(selectedCalendar == null) return;
+      CalendarAccess().addOrUpdateDates(listDateEntries, selectedCalendar);
+    }
+  });
+
+  //trigger ScheduleUpdatedCallback 10 seconds after the app started
+  Future.delayed(Duration(seconds: 10), () {
+    var scheduleSource = KiwiContainer().resolve<ScheduleSourceProvider>();
+    if (!scheduleSource.didSetupCorrectly()) return;
+    var scheduleProvider = KiwiContainer().resolve<ScheduleProvider>();
+    scheduleProvider.getUpdatedSchedule(
+      DateTime.now(),
+      DateTime.now().add(Duration(days: 30)),
+      CancellationToken(),
+    );
+  });
+
   isInitialized = true;
   print("Initialization finished");
 }
+
+class ListDateEntries30d {
+  List<DateEntry> listDateEntries;
+  ListDateEntries30d(this.listDateEntries) {}
+}
```

Comment:
Pack diese Klasse bitte auch noch in die neue `CalendarSynchronizer` Klasse mit rein, die ist dort besser aufghoben :)